tests/test_helpers.py

```diff
@@ -46,9 +45,9 @@ def test_parse_mimetype(mimetype, expected) -> None:
 
 # ------------------- guess_filename ----------------------------------
 
-def test_guess_filename_with_tempfile() -> None:
-    with tempfile.TemporaryFile() as fp:
-        assert (helpers.guess_filename(fp, 'no-throw') is not None)
+def test_guess_filename_with_tempfile(tmp_path) -> None:
+    file_path = tmp_path / 'test_guess_filename'
+    assert (helpers.guess_filename(file_path, 'no-throw') is not None)
```

Comment:
This is wrong: the original test was checking a file-like input and the update checks a path-like object input.
Don't change the original meaning of the tests please.

Reply:
I see, sorry about that. 

Would it be helpful to instead add other guess_filename tests?

Reply:
I corrected this to use `.open('w+b')`, which creates a file object in the same mode that `tempfile.TemporaryFile()` did.

Reply:
Yes. The extra test may be useful.

Reply:
OK -- added two tests (one for path, one a fallback to default).

Reply:
Added path as a separate test, as well as a test for default fallback, in 5928b41